services/src/main/java/org/fao/geonet/api/records/extent/WMTSBaseMapRenderer.java

```diff
@@ -0,0 +1,191 @@
+ * SRID2MatrixSet - (optional, default is no translation)
+ * flip4326 - (optional, default is NO)
+ * <p>
+ * Typically, you should just need to set the `wmtsGetCapabilitiesURL` and `layerName`.
+ * In this case, it will use the WMTS matrixSet with the same name as the SRS.
+ * <p>
+ * If you want to ALWAYS (!!) use a specific matrixSet, you can use the `matrixSet` parameter.
+ * <p>
+ * If the WMTS doesn't use the SRS as the name of the matrixSets, you can use the SRID2MatrixSet to convert the SRID
+ * to a matrixSet.
+ * <p>
+ * Most WMTS systems will use YX ordering for EPSG:4326.  If you are supplying coordinates in XY ordering, you will
+ * likely need to set flip4326 to true.
+ * <p>
+ * EXAMPLE
+ * <p>
+ * {
+ * "wmtsGetCapabilitiesURL":"URL",
+ * "layerName":"LAYERNAME",
+ * "SRID2MatrixSet": {
+ * "EPSG:900913":"EPSG:3857"
+ * },
+ * "flip4326": true
+ * }
+ * <p>
+ * A request for SRID EPSG:1234 will use the "EPSG:1234" matrixSet.
+ * A request for SRID EPSG:900913 will use the "EPSG:3857" matrixSet (SRID2MatrixSet).
+ * If the request is for EPSG:4326, the request bbox (AOI) will be flipped from XY to YX.
+ */
+public class WMTSBaseMapRenderer implements BaseMapRenderingEngine {
+
+
+    Envelope bbox;
+    String srs;
+    Dimension imageDimensions;
+    ServiceContext context;
+
+    URL getCapabilitiesURL;
+    String layerName;
+
+    Map<String, String> SRID2MatrixSet = new HashMap<>();
+    boolean flip4326 = false;
+
+    String urn4326 = "urn:ogc:def:crs:EPSG::4326"; // YX definition
+    /**
+     * null (not in config) -> matrix set is the same as srs
+     * otherwise, use this matrix set.
+     */
+    String matrixSet; //null -> use srs
+
+    public WMTSBaseMapRenderer() {
+
+    }
+
+    @Override
+    public boolean canHandle(String configString) {
+        return configString.startsWith("{");
+    }
+
+    public void configure(String configJsonString,
+                          Envelope bbox,
+                          String srs,
+                          Dimension imageDimensions,
+                          ServiceContext context) throws Exception {
+
+        if (configJsonString == null)
+            throw new Exception("GetMapBaseMapRenderer: null json/configuration");
+        if (bbox == null)
+            throw new Exception("GetMapBaseMapRenderer: null ubboxrl");
+        if (srs == null)
+            throw new Exception("GetMapBaseMapRenderer: null srs");
+        if (imageDimensions == null)
+            throw new Exception("GetMapBaseMapRenderer: null imageDimensions");
+
+
+        this.bbox = bbox;
+        this.srs = srs;
+        this.imageDimensions = imageDimensions;
+        this.context = context;
+
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> configuration = mapper.readValue(configJsonString, Map.class);
+
+        if (!configuration.containsKey("wmtsGetCapabilitiesURL")
```

Comment:
Maybe move `wmtsGetCapabilitiesURL` to a constant, check for other elements like `layerName`, `matrixSet` , etc.